### PR TITLE
feat: Add acceptance tests

### PR DESCRIPTION
**Pull request by zerok**
This implements acceptance tests using Dagger and Go tests.

This PR contains a lot of noise due to new Go modules . The relevant parts are in `dagger/main.go` and the `acceptance-tests` folder.

Resolves https://github.com/grafana/tanka/issues/220

### CODE REVIEW

File: dagger/README.md
Comment (iainlane) [inline dagger/README.md:1]: You could also add a summary of what the functions are doing in this file.

File: dagger/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Dagger setup for Tanka development
+
+This module includes dagger functions to be used during development of Dagger.
```
Comment (iainlane) [inline dagger/README.md:3]:
```suggestion
This module includes dagger functions to be used during development of Tanka.
```

File: .github/workflows/acceptance-tests.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+      -
+        name: Call Dagger Function
```
Comment (iainlane) [inline .github/workflows/acceptance-tests.yml:17]: also a nano nit

```suggestion
      - name: Call Dagger Function
```

**Review by iainlane — COMMENTED** [review]
cool!

File: dagger/main.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	return dag.Container().
+		Build(rootDir)
+}
+
+func (m *Tanka) AcceptanceTests(ctx context.Context, rootDir *Directory, acceptanceTestsDir *Directory) (string, error) {
+	// Determine Go version through Dockerfile
+	goVersion, err := dag.Container().
+		From("busybox").
+		WithMountedFile("/tmp/Dockerfile", rootDir.File("Dockerfile")).
+		WithExec([]string{"/bin/sh", "-c", "cat /tmp/Dockerfile | grep 'FROM golang' | grep 'as build' | awk -e '{print $2}'"}).
```
[the remark this answers is not in the document]
Comment (zerok) [reply dagger/main.go]: Done 🙂

File: acceptance-tests/apply_test.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		config: {
+	         apiVersion: "v1",
+	         kind: "ConfigMap",
+	         metadata : {
+	              name: "demo",
+	         },
+	         data: {},
+		},
+	}
+`
```
[the remark this answers is not in the document]
Comment (zerok) [reply acceptance-tests/apply_test.go]: done :)

File: acceptance-tests/apply_test.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestApplyEnvironment(t *testing.T) {
+	tmpDir := t.TempDir()
+	os.Chdir(tmpDir)
```
[the remark this answers is not in the document]
Comment (zerok) [reply acceptance-tests/apply_test.go]: Done 🙂

**Review by iainlane — APPROVED** [review]
tasty!

File: dagger/README.md
Comment (zerok) [reply dagger/README.md:1]: Done :)

**Pull request merged**